src/evaluators/DoWhileStatement.js

```diff
@@ -73,6 +73,7 @@ export default function(
   // so instead try to compute a fixpoint for it
   let iteration = () => {
     let bodyResult = env.evaluateCompletion(body, strictCode);
+    if (bodyResult instanceof Value) bodyResult = new NormalCompletion(bodyResult);
```

Comment:
I would have expected that evaluateCompletion will never return a Value and that this can be part of its type annotation.